backend/src/middlewares/roles-based-protection/role.middleware.ts

```diff
@@ -0,0 +1,36 @@
+import { TokenPayload } from '@autoline/shared';
+import { ExceptionMessage } from '@common/enums/exception/exception-message.enum';
+import { TypedRequestBody } from '@common/types/controller/controller';
+import { verifyToken } from '@helpers/token/token';
+import { Role } from '@prisma/client';
+import { Response, NextFunction } from 'express';
+import httpStatus from 'http-status-codes';
+
+const availableFor =
+  (roles: Role[]) =>
+  (
+    req: TypedRequestBody<{ tokenPayload: TokenPayload }>,
+    res: Response,
+    next: NextFunction,
+  ): void => {
+    let token = req.headers['authorization'];
+
+    if (token) {
+      token = token.replace('Bearer ', '');
+      const payload = verifyToken(token) as TokenPayload;
+
+      if (roles.includes(payload.role as Role)) {
+        req.body.tokenPayload = payload;
+        next();
+        return;
+      }
```

Comment:
This can be shortened to the:
```suggestion
      if (roles.includes(payload.role as Role)) {
        req.body.tokenPayload = payload;
        return next();
      }
```